Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -186,4 +192,79 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::transfer_to_fixed_length)
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // FIXME: 3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+
+    // 4. If IsDetachedBuffer(O) is true, return +0𝔽.
+    if (array_buffer_object->is_detached())
+        return Value(0);
+
+    size_t length;
+
+    // 5. If IsResizableArrayBuffer(O) is true, then
+    if (array_buffer_object->is_resizable()) {
+        // a. Let length be O.[[ArrayBufferMaxByteLength]].
+        length = array_buffer_object->max_byte_length().value();
+    }
+    // 6. Else,
+    else {
+        // a. Let length be O.[[ArrayBufferByteLength]].
+        length = array_buffer_object->byte_length();
+    }
+
+    // 7. Return 𝔽(length).
+    return Value(length);
+}
+
+// 1.3.3 get ArrayBuffer.prototype.resizable, https://tc39.es/proposal-resizablearraybuffer/#sec-get-arraybuffer.prototype.resizable
+JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::resizable_getter)
+{
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // FIXME: 3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+
+    // 4. Return IsResizableArrayBuffer(O).
+    return array_buffer_object->is_resizable();
+}
+
+// 1.3.5 ArrayBuffer.prototype.resize ( newLength ), https://tc39.es/proposal-resizablearraybuffer/#sec-arraybuffer.prototype.resize
+JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::resize)
+{
+    // 1. Let O be the this value.
+    auto array_buffer_object = TRY(typed_this_value(vm));
+
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferMaxByteLength]]).
+    if (!array_buffer_object->is_resizable())
+        return vm.throw_completion<TypeError>(ErrorType::ResizingNonResizableArray);
+
+    // FIXME: 3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+
+    // 4. If IsDetachedBuffer(O) is true, throw a TypeError exception.
+    if (array_buffer_object->is_detached())
+        return vm.throw_completion<TypeError>(ErrorType::DetachedArrayBuffer);
+
+    // 5. Let newByteLength be ? ToIntegerOrInfinity(newLength).
+    auto new_byte_length = TRY(vm.argument(0).to_integer_or_infinity(vm));
+
+    // NOTE: newByteLength < 0 is always false as new_byte_length is unsigned
+    // 6. If newByteLength < 0 or newByteLength > O.[[ArrayBufferMaxByteLength]], throw a RangeError exception.
+    if (new_byte_length < 0 || new_byte_length > array_buffer_object->max_byte_length().value())
+        return vm.throw_completion<RangeError>(ErrorType::InvalidByteLengthForResizableArrayBuffer, new_byte_length, array_buffer_object->max_byte_length().value());
+
+    // 7. Let hostHandled be ? HostResizeArrayBuffer(O, newByteLength)
+    TRY(host_resize_array_buffer(vm, array_buffer_object, new_byte_length));
+
+    // NOTE: There are more steps in the spec but we can always do the resizing internally, so this step will always return
```

Comment:
The very definition of host defined hooks is that they can customize behaviour behind your back, this is not an assumption we can make.